src/Groups.jl

```diff
@@ -4,6 +4,64 @@
 #
 ###############################################################################
 
+struct InfiniteOrder{T} <: Exception
+    x::T
+    InfiniteOrder(g::Union{GroupElem, Group}) = new{typeof(g)}(g)
+end
```

Comment:
This is out of scope for this PR, which is about getting rid of GroupsCore with minimal work. Any renaming or refactoring this enables for us can be done in a later commit, by anyone interested in investing time into it... But for now it'd be better to just get this done

Reply:
Actually I had a closer look and it turns out this is reference to as `GroupsCore.InfiniteOrder` everywhere, so we have to change those places, so I might as well rename it now.

Reply:
I wonder if we should export `InfiniteOrderError`...

Reply:
I would say yes. It would be strange, if some Oscar group (e.g. Weyl group of a generalized Cartan matrix) throws an `AbstractAlgebra.InfiniteOrderError` instead of an `InfiniteOrderError`.